api/helpers/gamestate/moves/resolve/ace.js

```diff
@@ -0,0 +1,28 @@
+module.exports = {
+  friendlyName: 'Resolve Ace One-Off',
+
+  description: 'Returns new GameState resulting from resolving an ace, which scraps all points on the field, and all jacks attached to those point cards.',
+
+  inputs: {
+    currentState: {
+      type: 'ref',
+      description: 'The latest GameState before the ace scraps all point cards and their attachments',
+      required: true,
+    },
+  },
+  sync: true, // synchronous helper
+  fn: ({ currentState }, exits) => {
+    let result = _.cloneDeep(currentState);
+
+    // Scrap all points and all their attachments
+    for (let card of [ ...currentState.p0.points, ...currentState.p1.points]) {
+      result.scrap.push(card, ...card.attachments);
+      card.attachments = [];
+    }
+
+    result.p0.points = [];
+    result.p1.points = [];
```

Comment:
I'm optimistic that similar to the result of splitting all the moves into their own helpers, that having separate one-off resolutions in their own helpers will make the actual logic of each resolving helper fairly lean. I'm pleased with how this one turned out.

It's notable that all the generic things about changing data e.g. updating the turn, scrapping the one-off itself and all the twos, setting the phase back to MAIN, all happens higher up the control flow in `resolve/execute.js` so each one off resolution is really only responsible for doing it's specific effect, without the general one-off cleanup